frame/im-online/src/tests.rs

```diff
@@ -96,7 +97,8 @@ fn should_report_offline_validators() {
 		advance_session();
 
 		// then
-		let offences = OFFENCES.with(|l| l.replace(vec![]));
+		let offences = OFFENCES::get();
```

Comment:
yeah, a `take` function is useful.

Reply:
My issue with implementing a `take` function on the macro is whether or not all types implement `Default::Defaut()`

Reply:
Seen your implementation. Will implement. Thought there would be issues.